Allow action types to perform their own mustache variable escaping in parameter templates

pmuellr opened the pull request:
resolves https://github.com/elastic/kibana/issues/79371
resolves https://github.com/elastic/kibana/issues/62928

In this PR, we allow action types to determine how to escape the
variables used in their parameters, when rendered as mustache
templates.  Prior to this, action parameters were recursively
rendered as mustache templates using the default mustache
templating, by the alerts library.  The default mustache
templating used html escaping.

Action types opt-in to the new capability via a new optional
method in the action type, `renderParameterTemplates()`.  If not
provided, the previous recursive rendering is done, but now with
no escaping at all.

For #62928, changed the mustache template rendering to be
replaced with the error message, if an error occurred,
so at least you can now see that an error occurred.  Useful
to diagnose problems with invalid mustache templates.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

## Release note

Improves the escaping done by the mustache templating engine, which previously always performed HTML escaping.  Escaping is now off for most action parameters, except those that need per-action escaping, including the Slack, Email and Webhook action parameters.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/)
* Commit: 4588f996764c607f2b685ebb1c7a6d246cb1f892
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/execution/node/365/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/","number":"89248","commit":"4588f996764c607f2b685ebb1c7a6d246cb1f892"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/)
* Commit: a9ba26525df97af86abd1bd9e2ad14cda021c685
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup9](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/execution/node/660/log)
* [Execute xpack-kibana-ciGroup9](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/execution/node/748/log)

### History
* :broken_heart: [Build #89248](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/) failed 4588f996764c607f2b685ebb1c7a6d246cb1f892

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/","number":"89248","commit":"4588f996764c607f2b685ebb1c7a6d246cb1f892"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/","number":"89849","commit":"a9ba26525df97af86abd1bd9e2ad14cda021c685"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/)
* Commit: 2a24ca990e43726fd5a0b2d86f099029f41a9c9e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/bb023c5c1c0225bc4062c036747e06472d49d508) | [after](https://github.com/elastic/kibana/commit/2a24ca990e43726fd5a0b2d86f099029f41a9c9e) | diff |
| --- | --- | --- | --- |
| `default` | 43058 | 43059 | +1 |

### History
* :broken_heart: [Build #89849](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/) failed a9ba26525df97af86abd1bd9e2ad14cda021c685
* :broken_heart: [Build #89248](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/) failed 4588f996764c607f2b685ebb1c7a6d246cb1f892

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/","number":"89248","commit":"4588f996764c607f2b685ebb1c7a6d246cb1f892"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/","number":"89849","commit":"a9ba26525df97af86abd1bd9e2ad14cda021c685"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/","number":"89861","commit":"2a24ca990e43726fd5a0b2d86f099029f41a9c9e"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/)
* Commit: da2fb13ce64f87db91703250a389ad72434ccad1

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/9ee1ec7f30b87c6c836fd378c97e0b265f400259) | [after](https://github.com/elastic/kibana/commit/da2fb13ce64f87db91703250a389ad72434ccad1) | diff |
| --- | --- | --- | --- |
| `default` | 43019 | 43020 | +1 |

### History
* :green_heart: [Build #89861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/) succeeded 2a24ca990e43726fd5a0b2d86f099029f41a9c9e
* :broken_heart: [Build #89849](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/) failed a9ba26525df97af86abd1bd9e2ad14cda021c685
* :broken_heart: [Build #89248](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/) failed 4588f996764c607f2b685ebb1c7a6d246cb1f892

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/","number":"89248","commit":"4588f996764c607f2b685ebb1c7a6d246cb1f892"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/","number":"89849","commit":"a9ba26525df97af86abd1bd9e2ad14cda021c685"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/","number":"89861","commit":"2a24ca990e43726fd5a0b2d86f099029f41a9c9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/","number":"90099","commit":"da2fb13ce64f87db91703250a389ad72434ccad1"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/)
* Commit: ef2410da76f9d72b91778cf304a6b227b44a71e2

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/9ee1ec7f30b87c6c836fd378c97e0b265f400259) | [after](https://github.com/elastic/kibana/commit/ef2410da76f9d72b91778cf304a6b227b44a71e2) | diff |
| --- | --- | --- | --- |
| `default` | 43019 | 43020 | +1 |

### History
* :green_heart: [Build #90099](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/) succeeded da2fb13ce64f87db91703250a389ad72434ccad1
* :green_heart: [Build #89861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/) succeeded 2a24ca990e43726fd5a0b2d86f099029f41a9c9e
* :broken_heart: [Build #89849](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/) failed a9ba26525df97af86abd1bd9e2ad14cda021c685
* :broken_heart: [Build #89248](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/) failed 4588f996764c607f2b685ebb1c7a6d246cb1f892

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/","number":"89248","commit":"4588f996764c607f2b685ebb1c7a6d246cb1f892"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/","number":"89849","commit":"a9ba26525df97af86abd1bd9e2ad14cda021c685"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/","number":"89861","commit":"2a24ca990e43726fd5a0b2d86f099029f41a9c9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/","number":"90099","commit":"da2fb13ce64f87db91703250a389ad72434ccad1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/","number":"90133","commit":"ef2410da76f9d72b91778cf304a6b227b44a71e2"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/)
* Commit: 3e4e100e887cbea9890ee9d170c60bd2151436a6

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/9ee1ec7f30b87c6c836fd378c97e0b265f400259) | [after](https://github.com/elastic/kibana/commit/3e4e100e887cbea9890ee9d170c60bd2151436a6) | diff |
| --- | --- | --- | --- |
| `default` | 43019 | 43021 | +2 |

### History
* :green_heart: [Build #90133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/) succeeded ef2410da76f9d72b91778cf304a6b227b44a71e2
* :green_heart: [Build #90099](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/) succeeded da2fb13ce64f87db91703250a389ad72434ccad1
* :green_heart: [Build #89861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/) succeeded 2a24ca990e43726fd5a0b2d86f099029f41a9c9e
* :broken_heart: [Build #89849](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/) failed a9ba26525df97af86abd1bd9e2ad14cda021c685
* :broken_heart: [Build #89248](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/) failed 4588f996764c607f2b685ebb1c7a6d246cb1f892

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89248/","number":"89248","commit":"4588f996764c607f2b685ebb1c7a6d246cb1f892"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/","number":"89849","commit":"a9ba26525df97af86abd1bd9e2ad14cda021c685"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/","number":"89861","commit":"2a24ca990e43726fd5a0b2d86f099029f41a9c9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/","number":"90099","commit":"da2fb13ce64f87db91703250a389ad72434ccad1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/","number":"90133","commit":"ef2410da76f9d72b91778cf304a6b227b44a71e2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/","number":"91854","commit":"3e4e100e887cbea9890ee9d170c60bd2151436a6"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/)
* Commit: 0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/execution/node/636/log)
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/execution/node/766/log)

### History
* :green_heart: [Build #91854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/) succeeded 3e4e100e887cbea9890ee9d170c60bd2151436a6
* :green_heart: [Build #90133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/) succeeded ef2410da76f9d72b91778cf304a6b227b44a71e2
* :green_heart: [Build #90099](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/) succeeded da2fb13ce64f87db91703250a389ad72434ccad1
* :green_heart: [Build #89861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/) succeeded 2a24ca990e43726fd5a0b2d86f099029f41a9c9e
* :broken_heart: [Build #89849](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/) failed a9ba26525df97af86abd1bd9e2ad14cda021c685

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89849/","number":"89849","commit":"a9ba26525df97af86abd1bd9e2ad14cda021c685"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/","number":"89861","commit":"2a24ca990e43726fd5a0b2d86f099029f41a9c9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/","number":"90099","commit":"da2fb13ce64f87db91703250a389ad72434ccad1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/","number":"90133","commit":"ef2410da76f9d72b91778cf304a6b227b44a71e2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/","number":"91854","commit":"3e4e100e887cbea9890ee9d170c60bd2151436a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/","number":"91860","commit":"0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/)
* Commit: 881c3d26b04ee3a74ebd4117356aed58514218de
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/execution/node/636/log)
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/execution/node/764/log)

### History
* :broken_heart: [Build #91860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/) failed 0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a
* :green_heart: [Build #91854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/) succeeded 3e4e100e887cbea9890ee9d170c60bd2151436a6
* :green_heart: [Build #90133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/) succeeded ef2410da76f9d72b91778cf304a6b227b44a71e2
* :green_heart: [Build #90099](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/) succeeded da2fb13ce64f87db91703250a389ad72434ccad1
* :green_heart: [Build #89861](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/) succeeded 2a24ca990e43726fd5a0b2d86f099029f41a9c9e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/89861/","number":"89861","commit":"2a24ca990e43726fd5a0b2d86f099029f41a9c9e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/","number":"90099","commit":"da2fb13ce64f87db91703250a389ad72434ccad1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/","number":"90133","commit":"ef2410da76f9d72b91778cf304a6b227b44a71e2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/","number":"91854","commit":"3e4e100e887cbea9890ee9d170c60bd2151436a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/","number":"91860","commit":"0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/","number":"92517","commit":"881c3d26b04ee3a74ebd4117356aed58514218de"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/)
* Commit: fbec6205648645180175629f97c2fe8ffb3c31e8
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/execution/node/378/log)

### History
* :broken_heart: [Build #92517](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/) failed 881c3d26b04ee3a74ebd4117356aed58514218de
* :broken_heart: [Build #91860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/) failed 0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a
* :green_heart: [Build #91854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/) succeeded 3e4e100e887cbea9890ee9d170c60bd2151436a6
* :green_heart: [Build #90133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/) succeeded ef2410da76f9d72b91778cf304a6b227b44a71e2
* :green_heart: [Build #90099](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/) succeeded da2fb13ce64f87db91703250a389ad72434ccad1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90099/","number":"90099","commit":"da2fb13ce64f87db91703250a389ad72434ccad1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/","number":"90133","commit":"ef2410da76f9d72b91778cf304a6b227b44a71e2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/","number":"91854","commit":"3e4e100e887cbea9890ee9d170c60bd2151436a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/","number":"91860","commit":"0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/","number":"92517","commit":"881c3d26b04ee3a74ebd4117356aed58514218de"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/","number":"92560","commit":"fbec6205648645180175629f97c2fe8ffb3c31e8"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/)
* Commit: 7df4fed55ba316979db25cd91dd6838effac0cea

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/1f75d0b79fa5ffda1936c798d3147b27b427f53f) | [after](https://github.com/elastic/kibana/commit/7df4fed55ba316979db25cd91dd6838effac0cea) | diff |
| --- | --- | --- | --- |
| `default` | 46934 | 47696 | +762 |

### History
* :broken_heart: [Build #92560](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/) failed fbec6205648645180175629f97c2fe8ffb3c31e8
* :broken_heart: [Build #92517](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/) failed 881c3d26b04ee3a74ebd4117356aed58514218de
* :broken_heart: [Build #91860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/) failed 0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a
* :green_heart: [Build #91854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/) succeeded 3e4e100e887cbea9890ee9d170c60bd2151436a6
* :green_heart: [Build #90133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/) succeeded ef2410da76f9d72b91778cf304a6b227b44a71e2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/90133/","number":"90133","commit":"ef2410da76f9d72b91778cf304a6b227b44a71e2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/","number":"91854","commit":"3e4e100e887cbea9890ee9d170c60bd2151436a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/","number":"91860","commit":"0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/","number":"92517","commit":"881c3d26b04ee3a74ebd4117356aed58514218de"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/","number":"92560","commit":"fbec6205648645180175629f97c2fe8ffb3c31e8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/","number":"92571","commit":"7df4fed55ba316979db25cd91dd6838effac0cea"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/)
* Commit: bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/62e79eec7011cb8ffa6f1c9626f4f989656ddad1) | [after](https://github.com/elastic/kibana/commit/bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a) | diff |
| --- | --- | --- | --- |
| `default` | 47694 | 47696 | +2 |

### History
* :green_heart: [Build #92571](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/) succeeded 7df4fed55ba316979db25cd91dd6838effac0cea
* :broken_heart: [Build #92560](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/) failed fbec6205648645180175629f97c2fe8ffb3c31e8
* :broken_heart: [Build #92517](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/) failed 881c3d26b04ee3a74ebd4117356aed58514218de
* :broken_heart: [Build #91860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/) failed 0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a
* :green_heart: [Build #91854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/) succeeded 3e4e100e887cbea9890ee9d170c60bd2151436a6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91854/","number":"91854","commit":"3e4e100e887cbea9890ee9d170c60bd2151436a6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/","number":"91860","commit":"0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/","number":"92517","commit":"881c3d26b04ee3a74ebd4117356aed58514218de"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/","number":"92560","commit":"fbec6205648645180175629f97c2fe8ffb3c31e8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/","number":"92571","commit":"7df4fed55ba316979db25cd91dd6838effac0cea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/","number":"92622","commit":"bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/)
* Commit: 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/5cc9bf82d2f64e059a367a35ac6e966963ef4b24) | [after](https://github.com/elastic/kibana/commit/4081c285fb7b660ed1094e58bd5cd2ddc06be8e9) | diff |
| --- | --- | --- | --- |
| `default` | 46948 | 47710 | +762 |

### History
* :green_heart: [Build #92622](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/) succeeded bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a
* :green_heart: [Build #92571](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/) succeeded 7df4fed55ba316979db25cd91dd6838effac0cea
* :broken_heart: [Build #92560](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/) failed fbec6205648645180175629f97c2fe8ffb3c31e8
* :broken_heart: [Build #92517](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/) failed 881c3d26b04ee3a74ebd4117356aed58514218de
* :broken_heart: [Build #91860](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/) failed 0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/91860/","number":"91860","commit":"0ebb47e949c6ef0c3c298fde689d1b74b73e0b6a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/","number":"92517","commit":"881c3d26b04ee3a74ebd4117356aed58514218de"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/","number":"92560","commit":"fbec6205648645180175629f97c2fe8ffb3c31e8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/","number":"92571","commit":"7df4fed55ba316979db25cd91dd6838effac0cea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/","number":"92622","commit":"bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/","number":"92751","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"}]}
PIPELINE-->

pmuellr commented on x-pack/plugins/actions/server/builtin_action_types/email.test.ts:
```diff
@@ -396,4 +396,37 @@ describe('execute()', () => {
     `);
   });
+
+  test('renders parameter templates as expected', async () => {
```
This is the first of a new set of tests for action types with optional `renderParameterTemplates()` methods.  They don't exhaustively test the escaping, as that is done in the render-specific code - it only tests that the escaping is basically happening as expected.

pmuellr commented on x-pack/plugins/actions/server/builtin_action_types/email.ts:
```diff
@@ -140,10 +141,23 @@ export function getActionType(params: GetActionTypeParams): EmailActionType {
+  variables: Record<string, unknown>
+): ActionParamsType {
+  return {
+    // most of the params need no escaping
```
good example of how to do the traditional no-escape templating first, then customize the escaping for specific parameters

pmuellr commented on x-pack/plugins/actions/server/lib/mustache_renderer.ts:
```diff
@@ -0,0 +1,102 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Mustache from 'mustache';
```
This module contains all of the specific escaping functions that we currently support.  I'm guessing the Slack and Markdown ones may need some tweaking for edge cases - especially Slack where "escaping" is not well defined.  But works for all the basic cases I've tried.

pmuellr commented on x-pack/plugins/actions/server/mocks.ts:
```diff
@@ -32,10 +31,20 @@ const createStartMock = () => {
   return mock;
 };
 
+// this is a default renderer that escapes nothing
```
Not a real mock :-).  But had to do this, as there was so much other code dependent on the templating, which would have required a LOT more non-trivial changes to other unrelated tests.  Function tests have been added to test a complete end-to-end of alerts firing actions and capturing the results of the action executions.

pmuellr commented on x-pack/plugins/actions/server/plugin.ts:
```diff
@@ -484,3 +492,17 @@ export class ActionsPlugin implements Plugin<Promise<PluginSetupContract>, Plugi
   }
 }
+
+export function renderActionParameterTemplates<Params extends ActionTypeParams = ActionTypeParams>(
```
Here's where the action types get to override the new default templating method, which does not do any escaping.

pmuellr commented on x-pack/plugins/alerts/server/manual_tests/action_param_templates.sh:
```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
```
This is a handy little test script to play with changing the value of variables (the alert name) via script or ad hoc in the UI, once the script creates the actions and alerts.

pmuellr commented on x-pack/plugins/alerts/server/task_runner/transform_action_params.ts:
```diff
@@ -40,30 +43,21 @@ export function transformActionParams({
-  // type signature for the customizer, but rather than pollute the customizer
-  // with casts, seemed better to just do it in one place, here.
-  return (result as unknown) as AlertActionParams;
+  // when the list of variables we pass in here changes,
```
This is where we moved the mustache templating out of alerts, and into actions.

pmuellr commented on x-pack/test/alerting_api_integration/common/fixtures/plugins/actions_simulators/server/slack_simulation.ts:
```diff
@@ -7,22 +7,41 @@
+  const messages: string[] = [];
+
   return http.createServer((request, response) => {
+    // return the messages that were posted to be remembered
```
extended the slack and webhook simulators, to capture the data sent in some requests, and allow it to be retrieved.  We use this in some new fully-cycle function tests that create actions and alerts and track that the payloads generated by the actions are escaped properly.

elasticmachine commented:
Pinging @elastic/kibana-alerting-services (Team:Alerting Services)

ymao1 reviewed (commented):
Looks good and looks like it's working as expected!

I have been trying to reproduce the webhook failure locally by using random characters and quotes in the alertName and then adding the alertName to the webhook body but have been unable to reproduce the failure. Do you have any suggestions for reproducing the failed action?

YulNaumenko commented:
> Looks good and looks like it's working as expected!
> 
> I have been trying to reproduce the webhook failure locally by using random characters and quotes in the alertName and then adding the alertName to the webhook body but have been unable to reproduce the failure. Do you have any suggestions for reproducing the failed action?

I think [this test](https://github.com/elastic/kibana/blob/4081c285fb7b660ed1094e58bd5cd2ddc06be8e9/x-pack/plugins/actions/server/builtin_action_types/webhook.test.ts#L377) cover the issue case enough. Maybe you can to reproduce by setting 'x"y' as an alert name and then add this param to the Webhook message?

YulNaumenko reviewed (approved):
LGTM

ymao1 commented:
> I think [this test](https://github.com/elastic/kibana/blob/4081c285fb7b660ed1094e58bd5cd2ddc06be8e9/x-pack/plugins/actions/server/builtin_action_types/webhook.test.ts#L377) cover the issue case enough. Maybe you can reproduce it by setting 'x"y' as an alert name and then add this param to the Webhook message?

I tried that (something like `foo"bar` for the alert name and the webhook POST was successful although i did see the quote sign turn into `&quot;`. I was just looking for a way to replicate the failed webhook POST

pmuellr commented:
> I have been trying to reproduce the webhook failure locally by using random characters and quotes in the alertName and then adding the alertName to the webhook body but have been unable to reproduce the failure. Do you have any suggestions for reproducing the failed action?

I assume you mean replicate the problems we saw before this PR?

I thought originally webhook would be problematic with double quote, but the old code escaped that as an html entity (as you saw).  The other problematic character is `\n` - the character.  aka `0x0a`.  The problem trying to simulate that with the manual test case I added is that there's no easy way to get a `0x0a` char in a string in JSON, as that makes the string unparseable (strings in JSON can't contain line breaks explicitly, but can include `\n`).  It needs to come from some place other than a JSON-sourced thing.

I was able to test this by hacking some bits while debugging, but agree it would be good to have an explicit test for this.  Just had a thought that we could extend one of our FT test alerts to add a context variable with an embedded `0x0a`, and then check that one in the new FT cases I added.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/)
* Commit: 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/execution/node/642/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/execution/node/755/log)

### History
* :green_heart: [Build #92751](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/) succeeded 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* :green_heart: [Build #92622](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/) succeeded bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a
* :green_heart: [Build #92571](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/) succeeded 7df4fed55ba316979db25cd91dd6838effac0cea
* :broken_heart: [Build #92560](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/) failed fbec6205648645180175629f97c2fe8ffb3c31e8
* :broken_heart: [Build #92517](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/) failed 881c3d26b04ee3a74ebd4117356aed58514218de

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92517/","number":"92517","commit":"881c3d26b04ee3a74ebd4117356aed58514218de"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/","number":"92560","commit":"fbec6205648645180175629f97c2fe8ffb3c31e8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/","number":"92571","commit":"7df4fed55ba316979db25cd91dd6838effac0cea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/","number":"92622","commit":"bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/","number":"92751","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/","number":"93501","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"}]}
PIPELINE-->

pmuellr commented:
> I was able to test this by hacking some bits while debugging, but agree it would be good to have an explicit test for this. Just had a thought that we could extend one of our FT test alerts to add a context variable with an embedded `0x0a`, and then check that one in the new FT cases I added.

Just added commit 651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec which provides a way to test some of these harder escapable strings, by having the alert inject context variables with escapable strings directly.

ymao1 reviewed (approved):
LGTM! Thanks for adding that test!

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/)
* Commit: 651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/execution/node/651/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/execution/node/758/log)

### History
* :broken_heart: [Build #93501](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/) failed 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* :green_heart: [Build #92751](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/) succeeded 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* :green_heart: [Build #92622](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/) succeeded bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a
* :green_heart: [Build #92571](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/) succeeded 7df4fed55ba316979db25cd91dd6838effac0cea
* :broken_heart: [Build #92560](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/) failed fbec6205648645180175629f97c2fe8ffb3c31e8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92560/","number":"92560","commit":"fbec6205648645180175629f97c2fe8ffb3c31e8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/","number":"92571","commit":"7df4fed55ba316979db25cd91dd6838effac0cea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/","number":"92622","commit":"bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/","number":"92751","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/","number":"93501","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/","number":"93687","commit":"651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93711/)
* Commit: da7b387c6a63c00d3e05ae890b861c55f13d22eb

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/ac189b8c251575fee1318b732cd19917992a38ec) | [after](https://github.com/elastic/kibana/commit/da7b387c6a63c00d3e05ae890b861c55f13d22eb) | diff |
| --- | --- | --- | --- |
| `default` | 46982 | 47744 | +762 |

### History
* :broken_heart: [Build #93687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/) failed 651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec
* :broken_heart: [Build #93501](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/) failed 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* :green_heart: [Build #92751](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/) succeeded 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* :green_heart: [Build #92622](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/) succeeded bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a
* :green_heart: [Build #92571](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/) succeeded 7df4fed55ba316979db25cd91dd6838effac0cea

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92571/","number":"92571","commit":"7df4fed55ba316979db25cd91dd6838effac0cea"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/","number":"92622","commit":"bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/","number":"92751","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/","number":"93501","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/","number":"93687","commit":"651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93711/","number":"93711","commit":"da7b387c6a63c00d3e05ae890b861c55f13d22eb"}]}
PIPELINE-->

pmuellr commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93877/)
* Commit: 50a80cd96095458c29b217f58067b59370c13fa3

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/ae64fc259222f1147c1500104d7dcb4cfa263b63) | [after](https://github.com/elastic/kibana/commit/50a80cd96095458c29b217f58067b59370c13fa3) | diff |
| --- | --- | --- | --- |
| `default` | 47129 | 47891 | +762 |

### History
* :green_heart: [Build #93711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93711/) succeeded da7b387c6a63c00d3e05ae890b861c55f13d22eb
* :broken_heart: [Build #93687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/) failed 651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec
* :broken_heart: [Build #93501](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/) failed 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* :green_heart: [Build #92751](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/) succeeded 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* :green_heart: [Build #92622](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/) succeeded bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92622/","number":"92622","commit":"bb4cf80b0ddbf519f1e9f7ab150143546aa53e6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/","number":"92751","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/","number":"93501","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/","number":"93687","commit":"651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93711/","number":"93711","commit":"da7b387c6a63c00d3e05ae890b861c55f13d22eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93877/","number":"93877","commit":"50a80cd96095458c29b217f58067b59370c13fa3"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94154/)
* Commit: 9e1496848047ce1b6f8d3dead476ceb88639d966

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/ae64fc259222f1147c1500104d7dcb4cfa263b63) | [after](https://github.com/elastic/kibana/commit/9e1496848047ce1b6f8d3dead476ceb88639d966) | diff |
| --- | --- | --- | --- |
| `default` | 47129 | 47891 | +762 |

### History
* :green_heart: [Build #93877](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93877/) succeeded 50a80cd96095458c29b217f58067b59370c13fa3
* :green_heart: [Build #93711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93711/) succeeded da7b387c6a63c00d3e05ae890b861c55f13d22eb
* :broken_heart: [Build #93687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/) failed 651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec
* :broken_heart: [Build #93501](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/) failed 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9
* :green_heart: [Build #92751](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/) succeeded 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92751/","number":"92751","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/","number":"93501","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/","number":"93687","commit":"651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93711/","number":"93711","commit":"da7b387c6a63c00d3e05ae890b861c55f13d22eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93877/","number":"93877","commit":"50a80cd96095458c29b217f58067b59370c13fa3"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94154/","number":"94154","commit":"9e1496848047ce1b6f8d3dead476ceb88639d966"}]}
PIPELINE-->

pmuellr commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94234/)
* Commit: 589c5d80abe9996a350246197750e0debd1da3e3

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/1f774bb2e62c2cde8840d29aa595e1c0889700d5) | [after](https://github.com/elastic/kibana/commit/589c5d80abe9996a350246197750e0debd1da3e3) | diff |
| --- | --- | --- | --- |
| `default` | 47145 | 47907 | +762 |

### History
* :green_heart: [Build #94154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94154/) succeeded 9e1496848047ce1b6f8d3dead476ceb88639d966
* :green_heart: [Build #93877](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93877/) succeeded 50a80cd96095458c29b217f58067b59370c13fa3
* :green_heart: [Build #93711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93711/) succeeded da7b387c6a63c00d3e05ae890b861c55f13d22eb
* :broken_heart: [Build #93687](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/) failed 651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec
* :broken_heart: [Build #93501](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/) failed 4081c285fb7b660ed1094e58bd5cd2ddc06be8e9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93501/","number":"93501","commit":"4081c285fb7b660ed1094e58bd5cd2ddc06be8e9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93687/","number":"93687","commit":"651eefcfebeafe46ed7cbe11f885b2b5b7c5d3ec"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93711/","number":"93711","commit":"da7b387c6a63c00d3e05ae890b861c55f13d22eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93877/","number":"93877","commit":"50a80cd96095458c29b217f58067b59370c13fa3"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94154/","number":"94154","commit":"9e1496848047ce1b6f8d3dead476ceb88639d966"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94234/","number":"94234","commit":"589c5d80abe9996a350246197750e0debd1da3e3"}]}
PIPELINE-->
